### PR TITLE
[12.0][MIG] stock_quant_manual_assign

### PR DESCRIPTION
**Pull request by LoisRForgeFlow**
@Eficent

### CODE REVIEW

File: stock_quant_manual_assign/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+# Copyright 2017 Eficent - Jordi Ballester
+# Copyright 2018 Fanha Giang
+# Copyright 2018 Tecnativa - Vicent Cubells
+# Copyright 2016-2018 Tecnativa - Pedro M. Baeza
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+{
+    "name": "Stock - Manual Quant Assignment",
+    "version": "12.0.1.0.0",
+    "category": "Warehouse Management",
```
Comment (rousseldenis) [inline stock_quant_manual_assign/__manifest__.py]: Change this to 'Warehouse'

File: stock_quant_manual_assign/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    "license": "AGPL-3",
+    "author": "AvanzOSC, "
+              "Tecnativa, "
+              "Eficent, "
+              "Fanha Giang, "
+              "Odoo Community Association (OCA)",
+    "website": "https://odoo-community.org",
```
Comment (rousseldenis) [inline stock_quant_manual_assign/__manifest__.py]: Change this to comply OCA

File: stock_quant_manual_assign/wizard/assign_manual_quants.py [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+
+    @api.multi
+    @api.constrains('quants_lines')
+    def _check_qty(self):
+        precision_digits = self.env[
+            'decimal.precision'].precision_get('Product Unit of Measure')
+        for record in self.filtered('quants_lines'):
+            if float_compare(record.lines_qty, record.move_id.product_qty,
+                             precision_digits=precision_digits) > 0:
+                raise UserError(
```
Comment (rousseldenis) [inline stock_quant_manual_assign/wizard/assign_manual_quants.py]: Please change to ValidationError

File: stock_quant_manual_assign/wizard/assign_manual_quants.py [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+        comodel_name='assign.manual.quants', string='Move', required=True,
+        ondelete='cascade')
+    quant_id = fields.Many2one(
+        comodel_name='stock.quant', string='Quant', required=True,
+        ondelete='cascade', oldname='quant')
+    location_id = fields.Many2one(
+        comodel_name='stock.location',
+        string='Location',
+        related='quant_id.location_id',
+        readonly=True,
```
Comment (rousseldenis) [inline stock_quant_manual_assign/wizard/assign_manual_quants.py]: Not needed anymore (same under)

File: stock_quant_manual_assign/wizard/assign_manual_quants.py [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+                lambda ml: (ml.location_id == quant.location_id and
+                            ml.lot_id == quant.lot_id)
+            )
+            reserved = (
+                quant.reserved_quantity - sum(
+                    move_lines.mapped('product_uom_qty'))
+            )
+            if float_compare(record.qty, record.quant_id.quantity - reserved,
+                             precision_digits=precision_digits) > 0:
+                raise UserError(
```
Comment (rousseldenis) [inline stock_quant_manual_assign/wizard/assign_manual_quants.py]: Please change to ValidationError

**Review by rousseldenis — CHANGES_REQUESTED** [review]
Code review. LGTM. Some comments.

File: stock_quant_manual_assign/wizard/assign_manual_quants.py (hunk 5 above)
Comment (LoisRForgeFlow) [reply stock_quant_manual_assign/wizard/assign_manual_quants.py]: Everything attended!

**Review by rousseldenis — APPROVED** [review]

**Review by HviorForgeFlow — APPROVED** [review]
LGTM :+1:
Functional review

**Comment by LoisRForgeFlow** [comment]
Thanks for you reviews! :smiley:. Now merging.

**Review by AaronHForgeFlow — APPROVED** [review]
:+1: LGTM

**Pull request closed without merging**